SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp

```diff
@@ -1,9 +1,859 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 
 #include "TestDefinitions.h"
+#include "SpatialGDKEditorSchemaGenerator.h"
+#include "SpatialGDKServicesModule.h"
+#include "SchemaGenObjectStub.h"
+#include "ExpectedGeneratedSchemaFileContents.h"
+
+#include "Utils/SchemaDatabase.h"
 
 #include "CoreMinimal.h"
+#include "HAL/PlatformFilemanager.h"
+#include "Misc/FileHelper.h"
 
 #define SCHEMA_GENERATOR_TEST(TestName) \
 	TEST(SpatialGDKEditor, SchemaGenerator, TestName)
 
+namespace
+{
+const FString gSchemaOutputFolder = FPaths::Combine(FSpatialGDKServicesModule::GetSpatialOSDirectory(), TEXT("Tests/"));
+const FString gSchemaDatabaseFileName = TEXT("Spatial/Tests/SchemaDatabase");
+const FString gDatabaseOutputFile = TEXT("/Game/Spatial/Tests/SchemaDatabase");
+
+TArray<FString> LoadSchemaFileForClassToStringArray(const FString& SchemaOutputFolder, const UClass* CurrentClass)
+{
+	FString SchemaFileFolder = TEXT("");
+
+	if (!CurrentClass->IsChildOf<AActor>())
+	{
+		SchemaFileFolder = TEXT("Subobjects");
+	}
+
+	TArray<FString> FileContent;
+	FFileHelper::LoadFileToStringArray(FileContent, *FPaths::SetExtension(FPaths::Combine(FPaths::Combine(SchemaOutputFolder, SchemaFileFolder), CurrentClass->GetName()), TEXT(".schema")));
+
+	return FileContent;
+}
+
+struct NamesAndIds
+{
+	TArray<FString> Names;
+	TArray<int32> Ids;
+};
```

Comment:
Maybe `ComponentNames` and `ComponentIDs`?

Reply:
How about changing the struct name to `ComponentNamesAndIds`?

Reply:
Renamed to `ComponentNamesAndIds`